Massif fails to create connector after a failed import/export

**Behavior**: in an Eclipse with Massif installed if an import/export operation fails, it must be restarted, otherwise, further operations can't be initiated because an error message is shown saying "Failed to create MatlabEngine".

**Steps to reproduce**: a way of reproducing the error is to try exporting a model that has a block with an invalid parameter value.

**Stacktrace**: 
```
!ENTRY hu.bme.mit.massif.simulink.ui 4 0 2018-08-05 14:45:48.205
!MESSAGE Exception occured when creating Massif Matlab connector
!STACK 0
hu.bme.mit.massif.communication.ConnectorCreationException: Failed to create MatlabEngine
	at hu.bme.mit.massif.communication.matlabengine.MatlabEngineEvaluatorFactory.createCommandEvaluator(MatlabEngineEvaluatorFactory.java:33)
	at hu.bme.mit.massif.communication.manager.CommandEvaluatorManager.getCommandEvaluator(CommandEvaluatorManager.java:45)
	at hu.bme.mit.massif.simulink.ui.handlers.AbstractSimulinkHandler.prepareCommandEvaluator(AbstractSimulinkHandler.java:65)
	at hu.bme.mit.massif.simulink.ui.handlers.AbstractSimulinkHandler.getCommandEvaluator(AbstractSimulinkHandler.java:103)
	at hu.bme.mit.massif.simulink.ui.handlers.ExportModelHandler.execute(ExportModelHandler.java:60)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:291)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:93)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:55)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:318)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:252)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:161)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:152)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:494)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:487)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:204)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:433)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:446)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:472)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5686)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1370)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4940)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4518)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1170)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1059)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:667)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:597)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:152)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:656)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:592)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1498)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1471)
Caused by: com.mathworks.engine.EngineException: Unable to connect to MATLAB session "MATLAB_29758".
	at com.mathworks.engine.MatlabEngine.connectMatlab(MatlabEngine.java:264)
	at com.mathworks.engine.MatlabEngine.connectMatlab(MatlabEngine.java:243)
	at hu.bme.mit.massif.communication.matlabengine.MatlabEngineAccess.<init>(MatlabEngineAccess.java:27)
	at hu.bme.mit.massif.communication.matlabengine.MatlabEngineEvaluator.<init>(MatlabEngineEvaluator.java:35)
	at hu.bme.mit.massif.communication.matlabengine.MatlabEngineEvaluatorFactory.createCommandEvaluator(MatlabEngineEvaluatorFactory.java:31)
	... 48 more
```

--- a/plugins/hu.bme.mit.massif.communication.commandevaluation/src/hu/bme/mit/massif/communication/commandevaluation/CEServerCommandEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.commandevaluation/src/hu/bme/mit/massif/communication/commandevaluation/CEServerCommandEvaluatorFactory.java
@@ -37,7 +37,7 @@ public class CEServerCommandEvaluatorFactory implements ICommandEvaluatorFactory
     Map<String,ICommandEvaluator> evaluators = new HashMap<>();
     
 	@Override
-	public ICommandEvaluator createCommandEvaluator(
+	public ICommandEvaluator createOrGetCommandEvaluator(
 			Map<String, Object> parameters) throws ConnectorCreationException {
 		ICommandEvaluator evaluator = null;
 		

--- a/plugins/hu.bme.mit.massif.communication.commandevaluation/src/hu/bme/mit/massif/communication/commandevaluation/CEServerCommandEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.commandevaluation/src/hu/bme/mit/massif/communication/commandevaluation/CEServerCommandEvaluatorFactory.java
@@ -14,6 +14,7 @@ import hu.bme.mit.massif.communication.ConnectorCreationException;
 import hu.bme.mit.massif.communication.ICommandEvaluator;
 import hu.bme.mit.massif.communication.ICommandEvaluatorFactory;
 
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
@@ -33,16 +34,25 @@ public class CEServerCommandEvaluatorFactory implements ICommandEvaluatorFactory
     private static final String MATLAB_NOT_FOUND_WITH_GIVEN_PARAMETERS = "No running MATLAB server was found with the given parameters specified in Window/Preferences/Simulink Preferences";
     protected static final String MATLAB_RUNNING_CHECK_ERROR = "Exception occurred checking running Matlabs!";
 
+    Map<String,ICommandEvaluator> evaluators = new HashMap<>();
+    
 	@Override
 	public ICommandEvaluator createCommandEvaluator(
 			Map<String, Object> parameters) throws ConnectorCreationException {
-		ICommandEvaluator result = null;
+		ICommandEvaluator evaluator = null;
 		
 		// Get the required parameters from the parameter map and cast them to the right type
 		String hostAddress = (String) parameters.get("hostAddress");
 		int hostPort = (Integer) parameters.get("hostPort");
 		String serviceName = (String) parameters.get("serviceName");
 
+		// Synthetic ID derived from the provided session details
+		String sessionId = hostAddress + hostPort + serviceName;
+		// Check if there is an evaluator with the given <IP,PORT,SERVICE_NAME> triplet
+		if(evaluators.get(sessionId) != null) {
+		    return evaluators.get(sessionId);
+		}
+		
 		String osName = System.getProperty("os.name");
 		if (osName != null && osName.indexOf("win") >= 0) {
 		    // Check whether there is an open matlab session on this computer
@@ -55,12 +65,16 @@ public class CEServerCommandEvaluatorFactory implements ICommandEvaluatorFactory
             }
 		}
         try {
-            MatlabClient matlabClient = new MatlabClient(hostAddress, hostPort, serviceName);
-            result = new CommandEvaluatorImpl(matlabClient);
+            if (evaluator == null) {
+                MatlabClient matlabClient = new MatlabClient(hostAddress, hostPort, serviceName);
+                evaluator = new CommandEvaluatorImpl(matlabClient);
+                // Cache the evaluator
+                evaluators.put(sessionId, evaluator);
+            }
         } catch (MatlabRMIException e) {
             throw new ConnectorCreationException("Exception occurred while preparing command evaluator!", e);
         }
-        return result;
+        return evaluator;
 	}
 
     protected boolean checkExistingMatlabSession(String serviceName) throws ConnectorCreationException {

--- a/plugins/hu.bme.mit.massif.communication.commandevaluation/src/hu/bme/mit/massif/communication/commandevaluation/CEServerCommandEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.commandevaluation/src/hu/bme/mit/massif/communication/commandevaluation/CEServerCommandEvaluatorFactory.java
@@ -37,7 +37,7 @@ public class CEServerCommandEvaluatorFactory implements ICommandEvaluatorFactory
     Map<String,ICommandEvaluator> evaluators = new HashMap<>();
     
 	@Override
-	public ICommandEvaluator createOrGetCommandEvaluator(
+	public ICommandEvaluator getOrCreateCommandEvaluator(
 			Map<String, Object> parameters) throws ConnectorCreationException {
 		ICommandEvaluator evaluator = null;
 		

--- a/plugins/hu.bme.mit.massif.communication.localscript/src/hu/bme/mit/massif/communication/localscript/LocalScriptEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.localscript/src/hu/bme/mit/massif/communication/localscript/LocalScriptEvaluatorFactory.java
@@ -23,10 +23,15 @@ import hu.bme.mit.massif.communication.ICommandEvaluatorFactory;
  */
 public class LocalScriptEvaluatorFactory implements ICommandEvaluatorFactory {
 
-	@Override
+	private LocalScriptEvaluator evaluator;
+
+    @Override
 	public ICommandEvaluator createCommandEvaluator(
 			Map<String, Object> parameters) {
-	    return new LocalScriptEvaluator(new MatlabController());
+	    if(evaluator == null) {
+	        evaluator = new LocalScriptEvaluator(new MatlabController());	        
+	    }
+        return evaluator;
 	}
 
 	@Override

--- a/plugins/hu.bme.mit.massif.communication.localscript/src/hu/bme/mit/massif/communication/localscript/LocalScriptEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.localscript/src/hu/bme/mit/massif/communication/localscript/LocalScriptEvaluatorFactory.java
@@ -26,7 +26,7 @@ public class LocalScriptEvaluatorFactory implements ICommandEvaluatorFactory {
 	private LocalScriptEvaluator evaluator;
 
     @Override
-	public ICommandEvaluator createOrGetCommandEvaluator(
+	public ICommandEvaluator getOrCreateCommandEvaluator(
 			Map<String, Object> parameters) {
 	    if(evaluator == null) {
 	        evaluator = new LocalScriptEvaluator(new MatlabController());	        

--- a/plugins/hu.bme.mit.massif.communication.localscript/src/hu/bme/mit/massif/communication/localscript/LocalScriptEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.localscript/src/hu/bme/mit/massif/communication/localscript/LocalScriptEvaluatorFactory.java
@@ -26,7 +26,7 @@ public class LocalScriptEvaluatorFactory implements ICommandEvaluatorFactory {
 	private LocalScriptEvaluator evaluator;
 
     @Override
-	public ICommandEvaluator createCommandEvaluator(
+	public ICommandEvaluator createOrGetCommandEvaluator(
 			Map<String, Object> parameters) {
 	    if(evaluator == null) {
 	        evaluator = new LocalScriptEvaluator(new MatlabController());	        

--- a/plugins/hu.bme.mit.massif.communication.manager/src/hu/bme/mit/massif/communication/manager/CommandEvaluatorManager.java
+++ b/plugins/hu.bme.mit.massif.communication.manager/src/hu/bme/mit/massif/communication/manager/CommandEvaluatorManager.java
@@ -42,7 +42,7 @@ public enum CommandEvaluatorManager {
     	if(id != null){
     		ICommandEvaluatorFactory factory = getFactoryMap().get(id);
     		if(factory != null){
-    			result = factory.createOrGetCommandEvaluator(parameters);
+    			result = factory.getOrCreateCommandEvaluator(parameters);
     		}
     	}
     	return result;

--- a/plugins/hu.bme.mit.massif.communication.manager/src/hu/bme/mit/massif/communication/manager/CommandEvaluatorManager.java
+++ b/plugins/hu.bme.mit.massif.communication.manager/src/hu/bme/mit/massif/communication/manager/CommandEvaluatorManager.java
@@ -42,7 +42,7 @@ public enum CommandEvaluatorManager {
     	if(id != null){
     		ICommandEvaluatorFactory factory = getFactoryMap().get(id);
     		if(factory != null){
-    			result = factory.createCommandEvaluator(parameters);
+    			result = factory.createOrGetCommandEvaluator(parameters);
     		}
     	}
     	return result;

--- a/plugins/hu.bme.mit.massif.communication.matlabcontrol/src/hu/bme/mit/massif/communication/matlabcontrol/MatlabControlEvaluator.java
+++ b/plugins/hu.bme.mit.massif.communication.matlabcontrol/src/hu/bme/mit/massif/communication/matlabcontrol/MatlabControlEvaluator.java
@@ -21,8 +21,6 @@ import matlabcontrol.MatlabProxyFactoryOptions.Builder;
 
 /**
  * Class responsible for the low level operations with MATLAB
- * 
- * (The successor class of BasicOperationsApi utility class)
  */
 public class MatlabControlEvaluator extends AbstractCommandEvaluator<MatlabControlAccess> {
 		
@@ -33,6 +31,10 @@ public class MatlabControlEvaluator extends AbstractCommandEvaluator<MatlabContr
 		this.debugPrint = debugPrint;
     }
     
+	public void setDebugPrint(boolean debugPrint) {
+	    this.debugPrint = debugPrint; 
+	}
+	
     public MatlabControlEvaluator(MatlabProxy proxy) {
         super(new MatlabControlAccess(proxy));
     }

--- a/plugins/hu.bme.mit.massif.communication.matlabcontrol/src/hu/bme/mit/massif/communication/matlabcontrol/MatlabControlEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.matlabcontrol/src/hu/bme/mit/massif/communication/matlabcontrol/MatlabControlEvaluatorFactory.java
@@ -24,7 +24,7 @@ public class MatlabControlEvaluatorFactory implements ICommandEvaluatorFactory {
 	private MatlabControlEvaluator evaluator;
 
     @Override
-	public ICommandEvaluator createOrGetCommandEvaluator(
+	public ICommandEvaluator getOrCreateCommandEvaluator(
 			Map<String, Object> parameters) {
 		// Get the parameters
 		String matlabPath = (String) parameters.get("matlabPath");

--- a/plugins/hu.bme.mit.massif.communication.matlabcontrol/src/hu/bme/mit/massif/communication/matlabcontrol/MatlabControlEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.matlabcontrol/src/hu/bme/mit/massif/communication/matlabcontrol/MatlabControlEvaluatorFactory.java
@@ -24,7 +24,7 @@ public class MatlabControlEvaluatorFactory implements ICommandEvaluatorFactory {
 	private ICommandEvaluator evaluator;
 
     @Override
-	public ICommandEvaluator createCommandEvaluator(
+	public ICommandEvaluator createOrGetCommandEvaluator(
 			Map<String, Object> parameters) {
 		// Get the parameters
 		String matlabPath = (String) parameters.get("matlabPath");

--- a/plugins/hu.bme.mit.massif.communication.matlabcontrol/src/hu/bme/mit/massif/communication/matlabcontrol/MatlabControlEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.matlabcontrol/src/hu/bme/mit/massif/communication/matlabcontrol/MatlabControlEvaluatorFactory.java
@@ -21,16 +21,20 @@ import java.util.Map;
  */
 public class MatlabControlEvaluatorFactory implements ICommandEvaluatorFactory {
 
-	@Override
+	private ICommandEvaluator evaluator;
+
+    @Override
 	public ICommandEvaluator createCommandEvaluator(
 			Map<String, Object> parameters) {
 		// Get the parameters
 		String matlabPath = (String) parameters.get("matlabPath");
 		boolean debugPrint = (boolean) parameters.get("print_issued_commands");
 		
-		ICommandEvaluator result = new MatlabControlEvaluator(matlabPath, debugPrint);
-		
-        return result;
+        if (evaluator == null) {
+            evaluator = new MatlabControlEvaluator(matlabPath, debugPrint);
+        }
+
+        return evaluator;
 	}
 
 	@Override

--- a/plugins/hu.bme.mit.massif.communication.matlabcontrol/src/hu/bme/mit/massif/communication/matlabcontrol/MatlabControlEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.matlabcontrol/src/hu/bme/mit/massif/communication/matlabcontrol/MatlabControlEvaluatorFactory.java
@@ -21,7 +21,7 @@ import java.util.Map;
  */
 public class MatlabControlEvaluatorFactory implements ICommandEvaluatorFactory {
 
-	private ICommandEvaluator evaluator;
+	private MatlabControlEvaluator evaluator;
 
     @Override
 	public ICommandEvaluator createOrGetCommandEvaluator(
@@ -33,7 +33,9 @@ public class MatlabControlEvaluatorFactory implements ICommandEvaluatorFactory {
         if (evaluator == null) {
             evaluator = new MatlabControlEvaluator(matlabPath, debugPrint);
         }
-
+        
+        evaluator.setDebugPrint(debugPrint);
+        
         return evaluator;
 	}
 

--- a/plugins/hu.bme.mit.massif.communication.matlabengine/src/hu/bme/mit/massif/communication/matlabengine/MatlabEngineEvaluator.java
+++ b/plugins/hu.bme.mit.massif.communication.matlabengine/src/hu/bme/mit/massif/communication/matlabengine/MatlabEngineEvaluator.java
@@ -36,6 +36,10 @@ public class MatlabEngineEvaluator extends AbstractCommandEvaluator<MatlabEngine
 		this.debugPrint = debugPrint;
 	}
 
+    public void setDebugPrint(boolean debugPrint) {
+        this.debugPrint = debugPrint; 
+    }
+    
 	@Override
 	public IVisitableMatlabData evaluateCommand(String command, int nargout) {
 		return super.evaluateCommand(debugPrint ? command : command + ";", nargout);

--- a/plugins/hu.bme.mit.massif.communication.matlabengine/src/hu/bme/mit/massif/communication/matlabengine/MatlabEngineEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.matlabengine/src/hu/bme/mit/massif/communication/matlabengine/MatlabEngineEvaluatorFactory.java
@@ -26,7 +26,7 @@ public class MatlabEngineEvaluatorFactory implements ICommandEvaluatorFactory {
 	private MatlabEngineEvaluator evaluator;
 
     @Override
-	public ICommandEvaluator createOrGetCommandEvaluator(Map<String, Object> parameters) throws ConnectorCreationException {
+	public ICommandEvaluator getOrCreateCommandEvaluator(Map<String, Object> parameters) throws ConnectorCreationException {
 		boolean debugPrint = (boolean) parameters.get("print_issued_commands");
 		try {
             if (evaluator == null) {

--- a/plugins/hu.bme.mit.massif.communication.matlabengine/src/hu/bme/mit/massif/communication/matlabengine/MatlabEngineEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.matlabengine/src/hu/bme/mit/massif/communication/matlabengine/MatlabEngineEvaluatorFactory.java
@@ -23,7 +23,7 @@ import hu.bme.mit.massif.communication.ICommandEvaluatorFactory;
  */
 public class MatlabEngineEvaluatorFactory implements ICommandEvaluatorFactory {
 
-	private ICommandEvaluator evaluator;
+	private MatlabEngineEvaluator evaluator;
 
     @Override
 	public ICommandEvaluator createOrGetCommandEvaluator(Map<String, Object> parameters) throws ConnectorCreationException {
@@ -35,6 +35,7 @@ public class MatlabEngineEvaluatorFactory implements ICommandEvaluatorFactory {
         } catch (EngineException | InterruptedException e) {
 			throw new ConnectorCreationException("Failed to create " + getConnectorName(), e);
 		}
+		evaluator.setDebugPrint(debugPrint);
 		return evaluator;
 	}
 

--- a/plugins/hu.bme.mit.massif.communication.matlabengine/src/hu/bme/mit/massif/communication/matlabengine/MatlabEngineEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.matlabengine/src/hu/bme/mit/massif/communication/matlabengine/MatlabEngineEvaluatorFactory.java
@@ -26,7 +26,7 @@ public class MatlabEngineEvaluatorFactory implements ICommandEvaluatorFactory {
 	private ICommandEvaluator evaluator;
 
     @Override
-	public ICommandEvaluator createCommandEvaluator(Map<String, Object> parameters) throws ConnectorCreationException {
+	public ICommandEvaluator createOrGetCommandEvaluator(Map<String, Object> parameters) throws ConnectorCreationException {
 		boolean debugPrint = (boolean) parameters.get("print_issued_commands");
 		try {
             if (evaluator == null) {

--- a/plugins/hu.bme.mit.massif.communication.matlabengine/src/hu/bme/mit/massif/communication/matlabengine/MatlabEngineEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication.matlabengine/src/hu/bme/mit/massif/communication/matlabengine/MatlabEngineEvaluatorFactory.java
@@ -23,13 +23,16 @@ import hu.bme.mit.massif.communication.ICommandEvaluatorFactory;
  */
 public class MatlabEngineEvaluatorFactory implements ICommandEvaluatorFactory {
 
-	@Override
+	private ICommandEvaluator evaluator;
+
+    @Override
 	public ICommandEvaluator createCommandEvaluator(Map<String, Object> parameters) throws ConnectorCreationException {
-		ICommandEvaluator evaluator;
 		boolean debugPrint = (boolean) parameters.get("print_issued_commands");
 		try {
-			evaluator = new MatlabEngineEvaluator(debugPrint);
-		} catch (EngineException | InterruptedException e) {
+            if (evaluator == null) {
+                evaluator = new MatlabEngineEvaluator(debugPrint);
+            }
+        } catch (EngineException | InterruptedException e) {
 			throw new ConnectorCreationException("Failed to create " + getConnectorName(), e);
 		}
 		return evaluator;

--- a/plugins/hu.bme.mit.massif.communication/src/hu/bme/mit/massif/communication/ICommandEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication/src/hu/bme/mit/massif/communication/ICommandEvaluatorFactory.java
@@ -20,7 +20,7 @@ public interface ICommandEvaluatorFactory {
      * @param parameters a map containing the required parameters for the connection to the MATLAB session
      * @return the object with the ICommandEvaluator interface
      */
-	public ICommandEvaluator createOrGetCommandEvaluator(Map<String, Object> parameters) throws ConnectorCreationException;
+	public ICommandEvaluator getOrCreateCommandEvaluator(Map<String, Object> parameters) throws ConnectorCreationException;
 	
 	public String getConnectorName();
 	

--- a/plugins/hu.bme.mit.massif.communication/src/hu/bme/mit/massif/communication/ICommandEvaluatorFactory.java
+++ b/plugins/hu.bme.mit.massif.communication/src/hu/bme/mit/massif/communication/ICommandEvaluatorFactory.java
@@ -20,7 +20,7 @@ public interface ICommandEvaluatorFactory {
      * @param parameters a map containing the required parameters for the connection to the MATLAB session
      * @return the object with the ICommandEvaluator interface
      */
-	public ICommandEvaluator createCommandEvaluator(Map<String, Object> parameters) throws ConnectorCreationException;
+	public ICommandEvaluator createOrGetCommandEvaluator(Map<String, Object> parameters) throws ConnectorCreationException;
 	
 	public String getConnectorName();
 	

